test/beacon/Voucher.test.ts

```diff
@@ -626,29 +626,349 @@ describe('Voucher', function () {
+
+                // Claim
+                await expect(claimBoostOrClassic())
+                    .to.emit(voucher, 'TaskClaimedWithVoucher')
+                    .withArgs(taskId)
+                    .and.to.not.emit(voucherHub, 'VoucherRefunded');
+                const {
+                    voucherCreditBalance: voucherCreditBalancePostClaim,
+                    voucherRlcBalance: voucherRlcBalancePostClaim,
+                    requesterRlcBalance: requesterRlcBalancePostClaim,
+                } = await getVoucherAndRequesterBalances();
+                // Voucher credit and RLC balances should stay unchanged.
+                expect(voucherCreditBalancePostClaim)
+                    .to.be.equal(voucherCreditBalancePreClaim)
+                    .to.be.equal(voucherRlcBalancePostClaim)
+                    .to.be.equal(voucherRlcBalancePreClaim);
+                // Requester balance should increase.
+                expect(requesterRlcBalancePostClaim).to.be.equal(
+                    requesterRlcBalancePreClaim + taskPrice,
+                );
+                // Sponsored amount should stay unchanged.
+                expect(await voucher.getSponsoredAmount(dealId)).to.be.equal(dealSponsoredAmount);
+            }
+        });
+
+        describe('Should claim task when already claimed on PoCo', async () => {
+            const pocoClaim = () => iexecPocoInstance.claim(taskId);
+            const pocoClaimBoost = () => iexecPocoInstance.claimBoost(dealId, taskIndex);
+
+            it('Classic', async () => await runTest(voucherMatchOrders, pocoClaim, claim));
+            it('Boost', async () =>
+                await runTest(voucherMatchOrdersBoost, pocoClaimBoost, claimBoost));
 
-        // TODO when volume can be > 1
-        describe('Should claim task when deal is partially sponsored and sponsored amount is not divisible by volume', async () => {});
+            async function runTest(
+                matchOrdersBoostOrClassic: any,
+                pocoClaimBoostOrClassic: any,
+                claimBoostOrClassic: any,
+            ) {
+                await addEligibleAssets([app, dataset, workerpool]);
+                await matchOrdersBoostOrClassic();
+                const {
+                    voucherCreditBalance: voucherCreditBalancePreClaim,
+                    voucherRlcBalance: voucherRlcBalancePreClaim,
+                    requesterRlcBalance: requesterRlcBalancePreClaim,
+                } = await getVoucherAndRequesterBalances();
+                // The voucher should've fully sponsored the deal.
+                const dealSponsoredAmount = await voucher.getSponsoredAmount(dealId);
+                const taskSponsoredAmount = dealSponsoredAmount / volume;
+                expect(dealSponsoredAmount).to.be.equal(dealPrice);
+                expect(taskSponsoredAmount).to.be.equal(taskPrice);
+
+                // Claim task on PoCo.
+                await pocoClaimBoostOrClassic();
```

Comment:
Maybe assert claim did not revert or task is indeed failed before voucher claim

Reply:
[Check task status after poco claim](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/21/commits/41674c8d411b0de771d273509e5150b90886f5e7)